src/main/java/com/mathworks/ci/actions/RunMatlabBuildAction.java

```diff
@@ -0,0 +1,100 @@
+package com.mathworks.ci.actions;
+
+/**
+ * Copyright 2024, The MathWorks Inc.
+ *
+ */
+
+import java.io.File;
+import java.io.IOException;
+
+import hudson.FilePath;
+import hudson.model.Run;
+import hudson.console.LineTransformationOutputStream;
+
+import com.mathworks.ci.BuildArtifactAction;
+import com.mathworks.ci.BuildConsoleAnnotator;
+import com.mathworks.ci.MatlabExecutionException;
+import com.mathworks.ci.parameters.BuildActionParameters;
+import com.mathworks.ci.utilities.MatlabCommandRunner;
+
+public class RunMatlabBuildAction {
+    private BuildActionParameters params; 
+    private MatlabCommandRunner runner;
+    private BuildConsoleAnnotator annotator;
+
+    private static String DEFAULT_PLUGIN = 
+        "+ciplugins/+jenkins/getDefaultPlugins.m";
+    private static String BUILD_REPORT_PLUGIN = 
+        "+ciplugins/+jenkins/BuildReportPlugin.m";
+    private static String TASK_RUN_PROGRESS_PLUGIN = 
+        "+ciplugins/+jenkins/TaskRunProgressPlugin.m";
+
+    public RunMatlabBuildAction(MatlabCommandRunner runner, BuildConsoleAnnotator annotator, BuildActionParameters params) {
+        this.runner = runner;
+        this.annotator = annotator;
+        this.params = params;
+    }
+
+    public RunMatlabBuildAction(BuildActionParameters params) throws IOException, InterruptedException {
+        this(new MatlabCommandRunner(params), 
+                new BuildConsoleAnnotator(
+                    params.getTaskListener().getLogger(), 
+                    params.getBuild().getCharset()),
+                params);
+    }
+
+    public void run() throws IOException, InterruptedException, MatlabExecutionException {
+        // Copy plugins and override default plugins function
+        runner.copyFileToTempFolder(DEFAULT_PLUGIN, DEFAULT_PLUGIN);
+        runner.copyFileToTempFolder(BUILD_REPORT_PLUGIN, BUILD_REPORT_PLUGIN);
+        runner.copyFileToTempFolder(TASK_RUN_PROGRESS_PLUGIN, TASK_RUN_PROGRESS_PLUGIN);
+        
+        // Set environment variable
+        runner.addEnvironmentVariable(
+                "MW_MATLAB_BUILDTOOL_DEFAULT_PLUGINS_FCN_OVERRIDE",
+                "ciplugins.jenkins.getDefaultPlugins");
+
+        // Redirect output to the build annotator
+        runner.redirectStdOut(annotator);
+
+        // Prepare the build tool command
+        // TODO: Devise better solution then prepending the command
+        //   here.
+        String command = "addpath('" 
```

Comment:
I believe that the idea (both here and for the test action) is that we need some extra pieces of code available rather than just the script that contains the command that we're calling. In this case we need the build plugins to be on the path, and for the test action we need genscript.

In particular, we want the script that the user is calling to run in the directory that we launched MATLAB in, so we change back to that directory as the first line of the scratch script. However, after we `cd` back, we still need the function which overrides the default plugins to be on the path.

If you have any suggestions for other ways to resolve this definitely let me know!